File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/MediaResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.entity.Attachment;
import mastodon4j.entity.share.Response;

import java.io.File;
import java.io.InputStream;

/**
 * @author hecateball
 */
public interface MediaResource {

    /**
     * Uploading a media attachment.
     *
     * @param stream      input stream (media to be uploaded)
     * @param name        file name that uploading
     * @param description description of media
     * @return an Attachment that can be used when creating a status
     */
    public Response<Attachment> postMedia(InputStream stream, String name, String description);


    /**
     * Uploading a media attachment.
     *
     * @param file        file (media to be uploaded)
     * @param description description of media
     * @return an Attachment that can be used when creating a status
     */
    public Response<Attachment> postMedia(File file, String description);
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/MediaResource.kt
package work.socialhub.kmastodon.api

import work.socialhub.kmastodon.api.request.medias.MediasPostMediaRequest
import work.socialhub.kmastodon.api.response.Response
import work.socialhub.kmastodon.api.response.medias.MediasPostMediaResponse

interface MediaResource {

    /**
     * Uploading a media attachment.
     */
    fun postMedia(
        request: MediasPostMediaRequest
    ): Response<MediasPostMediaResponse>
}
